### ISSUE TITLE
TypeError: object of type 'NoneType' has no len()

### ISSUE THREAD

**Issue by mattl** [w3c/wptdashboard#491]
[nonetype.txt](https://github.com/w3c/wptdashboard/files/1752681/nonetype.txt)

**Comment by mattl** [comment]
https://gist.github.com/mattl/fd90eb66b49dde3b09176c0b90f65a10

**Comment by jugglinmike** [comment]
Thank you for the report. Can you please share some information about the conditions which prompted this behavior?

**Comment by mattl** [comment]
This is happening for me on the current version of run.py -- I've tried with an older (Feb 20) SHA -- 3023e579d6db6d66210b8d7f1df1f93b34b1a8e5 -- and the error is still occuring.

**Comment by jugglinmike** [comment]
When we landed gh-482, we made some of the `run.py` script's output available via the `--log INFO` option.

    2018-02-24 00:36:33,609 INFO   Running chunk 1 of 1 (attempt 1 of 3)
    2018-02-24 00:36:35,179 DEBUG   wpt:stdout Traceback (most recent call last):
    2018-02-24 00:36:35,179 DEBUG   wpt:stdout   File "./wpt", line 5, in <module>
    2018-02-24 00:36:35,180 DEBUG   wpt:stdout     wpt.main()
    2018-02-24 00:36:35,181 DEBUG   wpt:stdout   File "/home/wptrunner/web-platform-tests/tools/wpt/wpt.py", line 132, in main
    2018-02-24 00:36:35,182 DEBUG   wpt:stdout     rv = script(*args, **kwargs)
    2018-02-24 00:36:35,183 DEBUG   wpt:stdout   File "/home/wptrunner/web-platform-tests/tools/wpt/run.py", line 404, in run
    2018-02-24 00:36:35,183 DEBUG   wpt:stdout     **kwargs)
    2018-02-24 00:36:35,184 DEBUG   wpt:stdout   File "/home/wptrunner/web-platform-tests/tools/wpt/run.py", line 372, in setup_wptrunner
    2018-02-24 00:36:35,185 DEBUG   wpt:stdout     check_environ(kwargs["product"])
    2018-02-24 00:36:35,186 DEBUG   wpt:stdout   File "/home/wptrunner/web-platform-tests/tools/wpt/run.py", line 126, in check_environ
    2018-02-24 00:36:35,186 DEBUG   wpt:stdout     raise WptrunError(message)
    2018-02-24 00:36:35,187 DEBUG   wpt:stdout tools.wpt.run.WptrunError: Missing hosts file configuration. Run
    2018-02-24 00:36:35,187 DEBUG   wpt:stdout 
    2018-02-24 00:36:35,188 DEBUG   wpt:stdout python wpt make-hosts-file >> /etc/hosts
    2018-02-24 00:36:35,189 DEBUG   wpt:stdout 
    2018-02-24 00:36:35,189 DEBUG   wpt:stdout from a shell with Administrator privileges.
    2018-02-24 00:36:35,204 INFO   Return code from wptrunner: 1
    Traceback (most recent call last):
      File "./run/run.py", line 593, in <module>
        main(args.platform_id, platform, args, config)
      File "./run/run.py", line 226, in main
        chunk_test_count = len(expected_tests)
    TypeError: object of type 'NoneType' has no len()

The WPT CLI is exiting early without writing any logging data because the hosts file has not been set correctly.

**Comment by jugglinmike** [comment]
After some further investigation, I've found that my initial explanation isn't relevant. A misconfigured hosts file *will* produce the error that @mattl reported, but it wasn't the case in our case.

The root problem is that `run.py` crashes if the "raw log" produced by WPT CLI contains no test data. We were seeing this happen in two different cases:

- network time outs in Sauce Labs-mediated testing
- mismatches in installed browser versions

Although all of these situations reflect larger issues which will invalidate tests, `run.py` should definitely not crash. I've submitted gh-498 to correct that.
